src/librustc_target/spec/thumbv4t_nintendo_gba.rs

```diff
@@ -0,0 +1,80 @@
+//! Targets the Nintendo Game Boy Advance (GBA),
+//! a handheld game device from 2001.
+//!
+//! Please ping @Lokathor if changes are needed.
+//!
+//! The target profile assumes that you have the ARM binutils in your path (specifically the linker, `arm-none-eabi-ld`). They can be obtained for free for all major OSes from the ARM developer's website, and they may also be available in your system's package manager
+//!
+//! **Important:** This target profile **does not** specify a linker script or the ROM header. You'll still need to provide these yourself to construct a final  binary. Generally you'd do this with something like `-Clink-arg=-Tmy_script.ld` and `-Clink-arg=my_crt.o`.
+
+use crate::spec::{LinkerFlavor, PanicStrategy, RelocModel, Target, TargetOptions, TargetResult};
+
+pub fn target() -> TargetResult {
+    Ok(Target {
+        llvm_target: "thumbv4t-none-eabi".to_string(),
+        target_endian: "little".to_string(),
+        target_pointer_width: "32".to_string(),
+        target_c_int_width: "32".to_string(),
+        target_os: "none".to_string(),
+        target_env: "gba".to_string(),
+        target_vendor: "nintendo".to_string(),
+        arch: "arm".to_string(),
+        /* Data layout args are '-' separated:
+         * little endian
+         * stack is 64-bit aligned (EABI)
+         * pointers are 32-bit
+         * i64 must be 64-bit aligned (EABI)
+         * mangle names with ELF style
+         * native integers are 32-bit
+         * All other elements are default
+         */
+        data_layout: "e-S64-p:32:32-i64:64-m:e-n32".to_string(),
+        linker_flavor: LinkerFlavor::Ld,
+        options: TargetOptions {
+            linker: Some("arm-none-eabi-ld".to_string()),
+            linker_is_gnu: true,
+
+            // extra args passed to the external assembler
+            asm_args: vec!["-mcpu=arm7tdmi".to_string(), "-mthumb-interwork".to_string()],
+
+            cpu: "arm7tdmi".to_string(),
+
+            // minimum extra features, these cannot be disabled via -C
+            features: "+soft-float,+strict-align".to_string(),
+
+            executables: true,
+
+            relocation_model: RelocModel::Static,
+
+            //function_sections: bool,
+            //exe_suffix: String,
+            main_needs_argc_argv: false,
+
+            // if we have thread-local storage
+            has_elf_tls: false,
+
+            // don't have atomic compare-and-swap
+            atomic_cas: false,
+
+            // always just abort
+            panic_strategy: PanicStrategy::Abort,
+
+            // ABIs to not use
+            unsupported_abis: super::arm_base::unsupported_abis(),
+
+            // The minimum alignment for global symbols.
+            min_global_align: Some(4),
```

Comment:
This seems wrong for 64-bit values, for the same reason as the stack needs to be 8-byte aligned on function entry.

Reply:
ah, I wasn't clear on if that made everything at least 4 and then it could be higher if the type was higher aligned, or what.

Basically I don't want any globals to be _less than_ 4 aligned. The main reason is because `include_bytes!` puts `&[u8]` into your program, but for the DMA to work, those included byte slices (which are generally image data) should still start on a 4-byte bound.

But that can also be achieved other ways, so I'll cut that i suppose.

Reply:
Aah, yeah, this might be a moot point with `u64` since that has 8-byte alignment anyways.

Reply:
Yeah, nevermind me.